### DISCUSSION TITLE
Does `@editor` capability persist in all the chat? How to limit it only to messages with `@editor` command?

### DISCUSSION THREAD

**Discussion by antosha417** [olimorris/codecompanion.nvim#350] [Q&A]
As far as I understand as soon as I type `@editor` in the chat I enable the llm to edit my code on each prompt, is it right?

I would expect it to do so only when I explicitly write `@editor` in the prompt.
So I can chat a bit, than apply the edit, than chat some more and apply the edit again.

Doesn't it makes more sense?

**Comment by olimorris** [comment]
Once added to the chat, you've turned the LLM into an agent. From the README:

> In the plugin, tools are simply context that's given to an LLM via a system prompt. This gives it knowledge and a defined schema which it can include in its response for the plugin to parse, execute and feedback on. Essentially turning it into an Agent. Tools can be added as a participant to the chat buffer by using the @ key.

So it persists throughout the whole chat. Depending on which LLM you're using, it will only edit your code if you ask it too. I'd advise looking through the `chat.lua` file to understand how the plugin works generally.

Your expectation would require the history of the conversation with the LLM to be adjusted, i.e. deleting system prompts.
